src/utilities/effects.ts

```diff
@@ -5112,7 +5112,7 @@ const effects = {
     tagIds: number[],
     files: FileList,
     user: User | null,
-  ): Promise<PlanSlim | null> {
+  ): Promise<{ error?: Error; plan?: PlanSlim }> {
```

Comment:
A little pedantic/moot, but would it be possible to type this as `{ error: Error } | { plan: PlanSlim }` just to be clearer that it would never potentially return both an error and plan?

Reply:
Good suggestion.

Reply:
Though it makes the actual usage of the function a little tricker since you have to check for the existence of the `error` and `plan` properties in the returned object and can't just destructure the object like `const { error } = importPlan(...)`. If you try to do something like this:
```
const result = await importPlan(...)
if (result.error) {...}
```
you also get a TS error complaining that the property may not exist. `Property 'error' does not exist on type '{ error: Error; } | { plan: PlanSlim; }'.`

Reply:
Could alternatively always return an error message where an error is really indicated by a non-empty string. Or it could be null and we check for that.

Reply:
As in not try/catch-ing `importPlan` internally? Would work though almost all of our effects are try/caught so might be a little odd?

Reply:
True. I think I'll rescind the comment then. I do think that try/catching outside of `importPlan` is the easier to use, but it does look like we have cases where we return the `error` from the effect. I think I'm more in favor of consistency.